clients/vulnerable-code/src/main/kotlin/VulnerableCodeService.kt

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.clients.vulnerablecode
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.databind.json.JsonMapper
+import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+
+import okhttp3.OkHttpClient
+
+import retrofit2.Retrofit
+import retrofit2.converter.jackson.JacksonConverterFactory
+import retrofit2.http.Body
+import retrofit2.http.GET
+import retrofit2.http.POST
+import retrofit2.http.Path
+
+/**
+ * Interface for a REST service that allows interaction with the VulnerableCode API to query information about
+ * vulnerabilities detected for specific packages.
+ */
+interface VulnerableCodeService {
+    companion object {
+        /**
+         * Create a new service instance that connects to the [serverUrl] specified and uses the optionally provided
+         * [client].
+         */
+        fun create(serverUrl: String, client: OkHttpClient? = null): VulnerableCodeService {
+            val vcClient = client ?: OkHttpClient()
+            val retrofit = Retrofit.Builder()
+                .client(vcClient)
+                .baseUrl(serverUrl)
+                .addConverterFactory(JacksonConverterFactory.create(JsonMapper().registerKotlinModule()))
+                .build()
+
+            return retrofit.create(VulnerableCodeService::class.java)
+        }
+    }
+
+    /**
+     * Data class to represent a reference to a vulnerability. When querying the vulnerabilities for packages, each
+     * package in the result has a list of this type. In order to retrieve further details about vulnerabilities,
+     * an additional request has to performed.
+     */
+    data class HyperLinkedVulnerability(
+        /** A URL with information about the vulnerability. */
+        val url: String,
+
+        /** The ID of this vulnerability, typically a CVE identifier. */
+        @JsonProperty("vulnerability_id")
```

Comment:
Instead of these annotations, I'd propose to use the `SNAKE_CASE` naming strategy like we do in `FossIdRestService`:

https://github.com/oss-review-toolkit/ort/blob/4a262680bdf8db006ef897ee9cf2d50aaa9935b4/clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt#L46-L58